_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp

```diff
@@ -977,20 +977,39 @@ namespace
                                         bool isLowPower,
                                         const mfxFrameInfo& info)
     {
-        if (platform == MFX_HW_IVB || platform == MFX_HW_VLV)
+        if (platform <= MFX_HW_IVB || platform == MFX_HW_VLV)
+        {
             return 1;
+        }
         else if (!isLowPower)
         {
-            if((info.Width < 3840 && info.Height < 2160) ||
-               (info.PicStruct != MFX_PICSTRUCT_PROGRESSIVE))
+            if (platform <= MFX_HW_HSW_ULT)
             {
-                mfxU16 const DEFAULT_BY_TU[] = { 0, 8, 6, 4, 3, 2, 1, 1 };
-                return DEFAULT_BY_TU[targetUsage];
+                if ((info.Width < 3840 && info.Height < 2160) ||
```

Comment:
I have much simpler code change proposal: just drop _all_ code branches with <= HSW and go with the single default setting!